### ISSUE TITLE
DDP training hangs when `num_denoising > 0` and batch contains zero ground-truth targets

### ISSUE THREAD

**Issue by CoolinYu** [lyuwenyu/RT-DETR#672]
### Describe the bug

Multi-GPU DDP training hangs (GPU 100% util, no training progress) when `num_denoising > 0` and some GPUs receive a batch where all images have zero annotations. The hang occurs at the first `backward()` — DDP waits indefinitely for gradient synchronization.

### To Reproduce

1. Use a dataset with a high proportion of empty images (e.g., LEVIR-Ship, ~49% images have no objects)
2. Train with DDP (2+ GPUs)
3. Training hangs immediately, GPU util stays at 100%, no loss is printed

### Expected behavior

Training should proceed normally regardless of empty-image batches.

### Root Cause

In `get_contrastive_denoising_training_group()` (`src/zoo/rtdetr/denoising.py`), when all targets in the local batch are empty, the function returns early:

```python
max_gt_num = max(num_gts)
if max_gt_num == 0:
    return None, None, None, None
```

This causes different computation graphs across DDP processes:

| | GPU 0 (empty batch) | GPU 1 (has targets) |
|---|---|---|
| `dn_meta` | `None` | `{dn_positive_idx, ...}` |
| `dn_aux_outputs` in model output | **absent** | **present** |
| `denoising_class_embed` in graph | **not in graph** | **in graph** |
| gradient of `denoising_class_embed` | `None` (never computed) | computed |

DDP's gradient `all_reduce` waits for `denoising_class_embed`'s gradient from GPU 0, which never arrives → **deadlock**.

Setting `find_unused_parameters=True` does NOT resolve this in certain PyTorch versions due to the conditional branching in the forward pass.

### Confirmed by

- Single-GPU training: ✅ works
- Multi-GPU + `num_denoising: 0`: ✅ works
- Multi-GPU + AI-TOD dataset (few empty images): ✅ works
- Multi-GPU + LEVIR-Ship + `num_denoising: 0`: ✅ works
- Multi-GPU + LEVIR-Ship + default `num_denoising`: ❌ hangs

### Fix

**File**: `src/zoo/rtdetr/denoising.py`

**Change**: Instead of returning `None` when `max_gt_num == 0`, set `max_gt_num = 1` so the function continues to produce valid (but zero-contribution) denoising queries, keeping the computation graph identical across all GPUs.

```diff
     max_gt_num = max(num_gts)
+    is_empty_batch = False
     if max_gt_num == 0:
-        return None, None, None, None
+        max_gt_num = 1
+        is_empty_batch = True

     ...
     input_query_bbox = torch.zeros([bs, max_gt_num, 4], device=device)
+    if is_empty_batch:
+        input_query_bbox[..., :2] = 0.5   # cx, cy = image center
+        input_query_bbox[..., 2:] = 0.1   # w, h = small valid size
```

The dummy queries are masked by `pad_gt_mask=0`, contributing **zero loss** and thus **no impact on convergence**.

### Additional notes

The problem is most easily triggered with datasets having a high proportion of empty images (e.g., LEVIR-Ship: 1135/2320 = 48.9% empty). With batch_size=2 per GPU, ~24% of steps will have all-empty batches on one GPU, triggering the hang.
